Documentation/compatibility/wpf-accessibility-improvements-48.md

```diff
@@ -0,0 +1,205 @@
+
+1. __Dependency Properties__</br> New [DependencyProperties](xref:System.Windows.DependencyProperty) [SizeOfSet](xref:System.Windows.Automation.AutomationProperties.SizeOfSetProperty) and [PositionInSet](xref:System.Windows.Automation.AutomationProperties.PositionInSetProperty) have been added to the System.Windows.Automation.AutomationProperties namespace. A developer can set their values via XAML:
+```xml
+    <Button AutomationProperties.SizeOfSet="3" 
+            AutomationProperties.PositionInSet="1">Button 1</Button> 
+    <Button AutomationProperties.SizeOfSet="3" 
+            AutomationProperties.PositionInSet="2">Button 2</Button> 
+    <Button AutomationProperties.SizeOfSet="3" 
+            AutomationProperties.PositionInSet="3">Button 3</Button> 
+```
+
+
+2. __AutomationPeer virtual methods__</br> Virtual methods [GetSizeOfSetCore](xref:System.Windows.Automation.Peers.AutomationPeer.GetSizeOfSetCore) and [GetPositionInSetCore](xref:System.Windows.Automation.Peers.AutomationPeer.GetPositionInSetCore) have also been added to the [AutomationPeer](xref:System.Windows.Automation.Peers.AutomationPeer) class. A developer can provide values for SizeOfSet and PositionInSet by overriding these methods:
+```csharp
+    public class MyButtonAutomationPeer : ButtonAutomationPeer 
+    { 
+        protected override int GetSizeOfSetCore() 
+        { 
+            // Call into your own logic to provide a value for SizeOfSet 
+            return CalculateSizeOfSet(); 
+        } 
+
+         protected override int GetPositionInSetCore() 
+        { 
+            // Call into your own logic to provide a value for PositionInSet 
+            return CalculatePositionInSet(); 
+        } 
+    } 
+```
+3. __Automatic Values__<br/>Items in [ItemsControls](xref:System.Windows.Controls.ItemsControl) will provide a value for these properties automatically without additional action from the developer. If an ItemsControl is grouped, the collection of groups will be represented as a set and each group counted as a separate set, with each item inside that group providing it�s position inside that group as well as the size of the group. Automatic values are not affected by virtualization. Even if an item is not realized, it is still counted towards the total size of the set and affects the position in the set of it�s sibling items.
+
+##### ControllerFor property support
+UIAutomation�s ControllerFor property returns an array of automation elements that are manipulated by the automation element that supports this property. This property is commonly used for Auto-suggest accessibility. ControllerFor is used when an automation element affects one or more segments of the application UI or the desktop. Otherwise, it is hard to associate the impact of the control operation with UI elements. This feature adds the ability for controls to provide a value for ControllerFor property.
```

Comment:
```suggestion
UIAutomation’s ControllerFor property returns an array of automation elements that are manipulated by the automation element that supports this property. This property is commonly used for Auto-suggest accessibility. ControllerFor is used when an automation element affects one or more segments of the application UI or the desktop. Otherwise, it is hard to associate the impact of the control operation with UI elements. This feature adds the ability for controls to provide a value for the ControllerFor property.
```